Source/SwiftLintBuiltInRules/Rules/Lint/PrivateSwiftUIStatePropertyRule.swift

```diff
@@ -0,0 +1,157 @@
+                @State private var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                @State fileprivate var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                var isPlaying = false
+            }
+            """),
+            Example("""
+            struct ContentView: View {
+                @StateObject var foo = Foo()
+            }
+            """),
+            Example("""
+            struct Foo {
+                @State var bar = false
+            }
+            """),
+            Example("""
+            class Foo: ObservableObject {
+                @State var bar = Bar()
+            }
+            """),
+            Example("""
+            extension MyObject {
+                struct ContentView: View {
+                    @State private var isPlaying: Bool = false
+                }
+            }
+            """),
+            Example("""
+            actor ContentView: View {
+                @State private var isPlaying: Bool = false
+            }
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            struct ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            final class ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """),
+            Example("""
+            extension MyObject {
+                struct ContentView: View {
+                    @State ↓var isPlaying: Bool = false
+                }
+            }
+            """),
+            Example("""
+            actor ContentView: View {
+                @State ↓var isPlaying: Bool = false
+            }
+            """)
+        ]
+    )
+
+    func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
+
+private extension PrivateSwiftUIStatePropertyRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override var skippableDeclarations: [DeclSyntaxProtocol.Type] {
+            [ProtocolDeclSyntax.self]
+        }
+
+        override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+            guard let inheritedTypeCollection = node.inheritanceClause?.inheritedTypeCollection else {
+                return .skipChildren
+            }
+
+            return inheritedTypeCollection.conformsToViewProtocol ? .visitChildren : .skipChildren
+        }
+
+        override func visit(_ node: StructDeclSyntax) -> SyntaxVisitorContinueKind {
+            guard let inheritedTypeCollection = node.inheritanceClause?.inheritedTypeCollection else {
+                return .skipChildren
+            }
+
+            return inheritedTypeCollection.conformsToViewProtocol ? .visitChildren : .skipChildren
```

Comment:
Maybe we can reduce duplication by implementing this as
```suggestion
           node.inheritanceClause?.conformsToViewProtocol == true ? .visitChildren : .skipChildren
```

Reply:
Seems like a reasonable request to me, I'll move this logic to TypeInheritanceClauseSyntax